examples/chapter1/benchmark/benchmark_example3.cpp

```diff
@@ -0,0 +1,97 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *****************************************************************************/
+#include "sum_vector_data.h"
+#include <benchmark/benchmark.h>
+#include <klepsydra/core/cache_listener.h>
+#include <klepsydra/core/event_emitter_middleware_provider.h>
+
+// ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ----
+// Considerations:
+// 1) Vector resized and published are the same vector.
+// 2) Vector resized is previously created before EventEmitterMiddlewareProvider() call
+//    with the same size which will be requested to be resized.
+// 3) Ranges.
+//    1st range: poolsize of the smartpool.
+//    2nd range: vectorSize of the vector to be resized and published.
+//    3rd range: initialization bool. true-> initialize vector. false-> do not init vector.
+// ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ----
+static void BM_KCE_21_SmartPool_Test(benchmark::State &state)
+{
+    // Perform setup here
+    const int poolSize = state.range(0);
+    const int i_vectorSize(state.range(1));
+    const bool b_initialice = state.range(2);
+
+    // This vector initialization seems be deterministic for lambda's resize() to be
+    // done in EventEmitterMiddlewareProvider() initialization. Must be created with
+    // the same size as the one to be resized? or Do we create it with no size, and
+    // lambda's resize SHOULD DO WHOLE WORK???
+    std::vector<float> vf_vec(i_vectorSize);
+    //std::vector<float> vf_vec;
+    std::function<void(std::vector<float> &)> fx_initializerFunction;
+    if (b_initialice) {
+        fx_initializerFunction = ([&](std::vector<float> &vf_vector) {
+            vf_vector.resize(i_vectorSize);
+        });
+    } else {
+        fx_initializerFunction = nullptr;
+    }
+    kpsr::EventEmitterMiddlewareProvider<std::vector<float>> vectorProvider(nullptr,
+                                                                            "vector_provider",
+                                                                            poolSize,
+                                                                            fx_initializerFunction,
+                                                                            nullptr);
+    uint16_t ui_published_count = 0;
+    // This code gets timed
+    for (auto _ : state) {
+        vectorProvider.getPublisher()->publish(vf_vec);
+        ui_published_count++;
```

Comment:
nope. the counter increase should be done in the listener. This is very important